src/cli/Cli.ts

```diff
@@ -414,7 +427,44 @@ export class Cli {
       });
     }
 
-    return minimist(args, minimistOptions);
+    const rewrittenArgs = this.getRewrittenArgsList(args, commandInfo);
+
+    return minimist(rewrittenArgs, minimistOptions);
+  }
+
+  /**
+   * Rewrites arguments (if necessary) before passing them into minimist.
+   * Currently only boolean values are checked and fixed.
+   */
+  private getRewrittenArgsList(args: string[], commandInfo: CommandInfo | undefined): string[] {
+    const booleanTypes = commandInfo?.command.types.boolean || [];
+
+    if (booleanTypes.length === 0) {
+      return args;
+    }
+
+    // The list of arguments is checked for boolean values to rewrite. 
+    // Boolean values are arguments that:
+    //  - do not start with a dash (-) 
+    //  - have a preceding argument that occurs in the 'booleanTypes' list in it's long or short form.
+    return args.map((arg: string, index: number, array: string[]) => {
+      if (arg.startsWith("-") || index === 0) {
+        return arg;
+      }
+
+      if (booleanTypes.some(t => `--${t}` === array[index - 1] || `-${t}` === array[index - 1])) {
```

Comment:
Let's clarify this line with a comment: we're checking if the current arg is a value that belongs to a bool option

Reply:
I wrote the entire comment above the function, explaining this as well. But I'll move it. 👍

Reply:
Ah missed, it, sorry. Still, I think it would be clearer to add it above this line as well, as it's pretty cryptic 😊

Reply:
Yeah, I agree.

Reply:
> checking if the current arg is a value that belongs to a bool option

By the way, this is so much shorter and clearer, borrowing that 😂